### DISCUSSION TITLE
This how you use this plugin with a esp32 this may work for Arduino ble

### DISCUSSION THREAD

**Discussion by OP-EX** [dotnet-bluetooth-le/dotnet-bluetooth-le#729] [Show and tell]
string serviceId = " put the service Id here (Note the service Id  can be found in the code for the esp32/Arduino)";
  string characteristicId = put the characteristic Id here (Note the characteristic Id  can be found in the code for the esp32/Arduino)"; 

 IBluetoothLE BLE;
 IAdapter Adapter;
 IService Service;
 iCharacteristic Characteristic;
 List<IDevice> BluetoothDeviceList = new List<IDevice>();

    private void BluetoothDevices()
    {
        // this gets all the devices that are connected or paired to the device
        Adapter = CrossBluetoothLE.Current.Adapter;

        var SystemDevices = Adapter.GetSystemConnectedOrPairedDevices();

        foreach (var device in SystemDevices)
        {
            BluetoothDeviceList.Add(device);

        }

        Device_List.ItemsSource = BluetoothDeviceList; // This is for a listview in maui
    }

    private void OnConnectClicked(object sender, EventArgs e)
    {
        Connected();
    }
    private async void Connected()
    {
        //var systemDevices = Adapter.GetSystemConnectedOrPairedDevices();
        foreach (var device in BluetoothDeviceList)
        {
            await Adapter.ConnectToDeviceAsync(device);

            Connected_Device_Name.Text =  device.Name; // This is for a label to see the connected device
            

            Service = await device.GetServiceAsync(Guid.Parse(serviceId));
            Characteristic = await Service.GetCharacteristicAsync(Guid.Parse(characteristicId));

        }
    }
    private async void OnLed_TestClicked(object sender, EventArgs e)
    {
        var Led_Bool = false;
        Led_Bool = !Led_Bool;
        if (Led_Bool)
        {
            Led_Test.Text = "Led: On";
            await Characteristic.WriteAsync(Encoding.UTF8.GetBytes("n"));
        }
        else
        {
            Led_Test.Text = "Led: Off";
            await Characteristic.WriteAsync(Encoding.UTF8.GetBytes("f"));
        }
    }

**Comment by smsissuechecker** [comment]
Hi @OP-EX,

I'm the friendly issue checker.
It seems like (100.00 %) you haven't used our issue template :cry: I think it is very frustrating for the repository owners, if you ignore them.

If you think it's fine to make an exception, just ignore this message.
**But if you think it was a mistake to delete the template, please close the issue and create a new one.**

Thanks!
